src/main/java/edu/ucsb/cs56/ucsb_open_lab_scheduler/controllers/CourseOfferingController.java

```diff
@@ -113,6 +116,43 @@ public String deleteCourse(@PathVariable("id") long id, Model model, OAuth2Authe
             redirAttrs.addFlashAttribute("alertSuccess", "Course successfully deleted.");
         }
         return "redirect:/courseOffering/";
+    }
+
+    @GetMapping("/courseOffering/create")
+    public String create(CourseOffering courseOffering) {
+        return "courseOffering/create";
+    }
+
+    public boolean checkIfCourseExists(Iterable<CourseOffering> iter, CourseOffering courseOffering){
```

Comment:
**TLDR:** I don't think this function does what is intended, but I also don't think its intended behavior is necessary. Also there may be a way to achieve your intended behavior through a database query.

If you want to ensure that there is not a course already in the database with the same courseId/quarter/instructor/instructor email, this function will not be sufficient to catch duplicates because the `.equals` defined on the courseOffering entity also compares ids, which will always be unique because they are autogenerated. So you could create two courses through the form with identical inputs, and they would still not be equal.  However, we discussed that there might be a need to create two instances of a course with the same information in a given quarter for some reason, so we don't necessarily need to restrict this behavior. 

In the future, if you wanted to write a function like this, you could add create a custom query on the **repository** that searches for existing entries in the database whose fields match the fields of the object you are intending to add. For example:

`List<CourseOffering> findByCourseIdAndQuarterAndInstructorAndInstructorEmail(String courseId, String quarter, String instructor, String instructorEmail)`

Then you would be able to replace this function with a single database query, and simply check whether that query returned any results (ie if the returned list is empty).